site/content/concepts/command-and-control.md

```diff
@@ -3,29 +3,74 @@ title = "Command & Control"
 ## Command & Control over HTTP Adapter
 
-The following sequence diagram gives an overview of case 1. with a device connected via HTTP.
+The following sequence diagram gives an overview of a device connected via HTTP, which gets a command from the business application as soon as it gets online.
  
 ![Command & Control over HTTP Adapter](../command_control_concept_http.png) 
 
-With the `ttd`-[notification]({{< relref "device-notifications.md" >}}) in (1) the device indicates it will stay connected for max. 30 seconds. This notification will be consumed by the application (2) and it now tries to send a command (3) to the device at the given address `control/TENANT/4711` (If the device would not be connected and the adapter has not established a link for it, the application would got no credits so send the command).
+With the `hono-ttd`-[notification]({{< relref "device-notifications.md" >}}) in (1) the device indicates it will stay connected for max. 30 seconds. This notification will be consumed by the application (2) and it now tries to send a command (3) to the device at the given address `control/TENANT/4711` (If the device is not connected, the adapter has not established a link for it and the application would get no credits so send the command).
 
-The HTTP Adapter gets the command and writes it in the response of the devices `send event` (4).
+The HTTP Adapter gets the command and writes it in the response of the devices `send event` (4). The HTTP Adapter sets the following response headers and optionally a payload.
 
+| Response Header              | Description         |
+| :---------------------       |  :----------------- |
+| `hono-cmd`               | The command, which should be executed by the device. It could need further data from the payload. |
+| `hono-cmd-req-id`      | Id, which is needed in the command response to find the                     |
+
+ The `hono-cmd` is the command, which need to be known by the device. The device can handle it and may also use the payload. The `hono-cmd-req-id` is needed for the command response to correlate it. It will be send back from the device to the adapter in a following operation (5). 
+ 
 {{% note %}}
 This is, what is implemented in Hono 0.6. At the moment the Application does not get a response back from the device as shown in (5) and (6). 
 {{% /note %}}
+
+The device needs to respond to the command (5), to inform the business application about the success. It typically follows the way described in [HTTP Adapter]({{< relref "user-guide/http-adapter.md" >}}) for telemetry and events with the following changes:
+
+Authenticated Device:
+
+* URI: `/control/res` 
+* Method: `POST`
+
+Unauthenticated Device:
+
+* URI: `/control/res/${tenantId}/${deviceId}` 
+* Method: `PUT`
+
+Further:
+
+* Request Headers:
+  * `hono-cmd-req-id`: The value of the response-header (with same name) of the command event.
+* Request Body:
+  * (optional) Arbitrary payload as response data.
+* Status Codes: 
```

Comment:
are these the status codes contained in the response that the adapter sends back to the device or are these status codes that the device should include in its HTTP request to indicate the outcome of the command?

Reply:
The latter.